### ISSUE TITLE
API - update article not working

### ISSUE THREAD

**Issue by yafkari** [thepracticaldev/dev.to#4000]
**Describe the bug**

I was working on a little api client to update my posts from there, but then I realized that
the PUT method wasn't working (I think), I tried a POST request to be sure that wasn't from me.

The POST worked, it created a post.
But when I tried to send a PUT request, to update the title of an unpublished post, it didn't work.
I received a 200 status code, with the data of that post (like it's supposed to do if it worked), but the title is still unchanged.

Here is the code I used.
See Screenshot 1

I also tested the code that is present on the documentation but it isn't working too.
PS: Curly braces missing here ^^
See Screenshot 2

**To Reproduce**

Just try to send a PUT request to the dev.to api to modify an unpublished article (maybe it works, on published article, I didn't tried I think).

**Expected behavior**

Receive the 200 status code, and modify the title of my post.

**Screenshots**
1:
![1](https://user-images.githubusercontent.com/41365655/64641681-f8c70780-d40c-11e9-8574-5733ed275816.png)
2:
![2](https://user-images.githubusercontent.com/41365655/64641722-11cfb880-d40d-11e9-8c9d-f1c8adb64f2e.png)

**Comment by Zhao-Andy** [comment]
Hmm, I think we always render the 200 response and don't have error handling if the article does not update, based on:

https://github.com/thepracticaldev/dev.to/blob/590149fd48d87203ac2e164cc6c3ec6baf957b34/app/controllers/api/v0/articles_controller.rb#L42-L45

Might be wrong about this though. cc @rhymes 

If you're updating an article that's using front matter, you won't be able to update the title by updating the `title` attribute of the `article`. You'll have to update it with new body markdown that has a new title. For example:
```
# original markdown
---
title: something
published: false
# etc etc
---
some body text

# new markdown

---
title: some brand new title
published: false
# etc etc
---
some body text
```

This is the same with publishing the article as well. Articles that use front matter **must** update their front matter metadata via the `body_markdown` attribute.

**Comment by rhymes** [comment]
Hi @yafkari, how did you create the article? As @Zhao-Andy said, if there was a frontmatter you have to send the changed title in the frontmatter of the `body_markdown`. If instead you just didn't use a frontmatter, then we have a bug :)

@Zhao-Andy: you're correct, right now the API returns errors only if there were updating errors, it doesn't return errors if an update is overridden by the frontmatter.

**Comment by yafkari** [comment]
Thank you @Zhao-Andy and @rhymes for the quick reply.

@rhymes, I created the article with the editor v1, so with the frontmatter apparently.
I didn't know that I had to do it this way (with body_markdown), sorry for that.

So, apparently everything is fine ^^.

Honestly, it would be cool to be able to update an article the same way for every articles.

Anyway, thank you for the help and sorry for the false alarm !
